src/index.d.ts

```diff
@@ -0,0 +1,3 @@
+import UserConfig from "./UserConfig";
+
+export { UserConfig };
```

Comment:
Per #2089, do we want to reexport the `defaultConfig` as well?
```js
 * @typedef {import("@11ty/eleventy/src/UserConfig")} EleventyConfig
 * @typedef {ReturnType<import("@11ty/eleventy/src/defaultConfig")>} EleventyReturnValue
```

Reply:
That's a good question, I don't think so yet.  I was thinking of keeping the scope of this PR down to just setting up the initial type declaration file with the `UserConfig` (which is a well-defined class, making it a good initial type to export).  I was thinking exporting additional types can come after this, wanted to keep this one small, easier for @zachleat to review (and hopefully merge 😄 ).

BTW, getting that `EleventyReturnValue` brings up a few questions... should it return a `Partial` of that return type (since user config doesn't require all those types, they are all optional - they get merged with the defaultConfig)?  Should we make an actual explicit interface for `EleventyReturnValue` (the `ReturnType<typeof defaultConfig>` method of inferring types is missing some properties, like `dir.layouts`)?  Does this project want to have explicitly defined types/interfaces or just let typescript infer types based on the JS?  I personally like the "contract" of having explicitly defined types, but I'm just some random dude so I didn't want to include such decisions in this PR 😀